src/Tabs.php

```diff
@@ -57,86 +57,105 @@
  */
 final class Tabs extends Widget
 {
+    public const NAV_PILLS = 'nav-pills';
+
     private array $items = [];
-    private array $itemOptions = [];
-    private array $headerOptions = [];
-    private bool $encodeLabels = true;
     private bool $encodeTags = false;
     private string $navType = 'nav-tabs';
     private bool $renderTabContent = true;
     private array $tabContentOptions = [];
-    private string $dropdownClass = Dropdown::class;
+    private array $paneOptions = [];
     private array $panes = [];
-    private array $options = [];
+    private Nav $nav;
+    private array $navDefinitions = [];
 
-    protected function run(): string
+    public function getId(?string $suffix = '-tabs'): ?string
     {
-        if (!isset($this->options['id'])) {
-            $this->options['id'] = "{$this->getId()}-tabs";
-        }
+        return $this->navDefinitions['options']['id'] ?? parent::getId('-tabs');
+    }
 
-        /** @psalm-suppress InvalidArgument */
-        Html::addCssClass($this->options, ['widget' => 'nav', $this->navType]);
+    protected function beforeRun(): bool
+    {
         Html::addCssClass($this->tabContentOptions, ['tabContentOptions' => 'tab-content']);
 
-        $this->prepareItems($this->items);
-
-        $navWidget = Nav::widget()
-            ->dropdownClass($this->dropdownClass)
-            ->items($this->items)
-            ->options(ArrayHelper::merge(['role' => 'tablist'], $this->options));
+        $navDefinitions = $this->prepareNavDefinitions();
+        $this->prepareItems($this->items, $navDefinitions['options']['id']);
+        $this->nav = $this->prepareNav($navDefinitions, $this->items);
 
-        if ($this->encodeLabels === false) {
-            $navWidget = $navWidget->withoutEncodeLabels();
-        }
+        return parent::beforeRun();
+    }
 
-        return $navWidget->render() . $this->renderPanes($this->panes);
+    protected function run(): string
+    {
+        return $this->nav->render() . $this->renderPanes($this->panes);
     }
 
     /**
-     * Name of a class to use for rendering dropdowns withing this widget. Defaults to {@see Dropdown}.
+     * Set all options for nav widget
```

Comment:
Set all definitions for nav widget.

Reply:
> Set all definitions for nav widget.

[Done](https://github.com/yiisoft/yii-bootstrap5/pull/75)